src/util/config.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use rocksdb::DBCompressionType;
+
+pub fn get_compression_by_string(tp: &str) -> DBCompressionType {
+    match &*tp.to_owned().to_lowercase() {
+        "no" => DBCompressionType::DBNo,
+        "snappy" => DBCompressionType::DBSnappy,
+        "zlib" => DBCompressionType::DBZlib,
+        "bzip2" => DBCompressionType::DBBz2,
+        "lz4" => DBCompressionType::DBLz4,
+        "lz4hc" => DBCompressionType::DBLz4hc,
+        _ => panic!("unsupported compression type {}", tp),
+    }
+}
+
+fn split_property(property: &str) -> Result<(f64, &str), (())> {
+    let mut indx = 0;
+    for s in property.chars() {
+        match s {
+            '0'...'9' | '.' => {
+                indx += 1;
+            }
+            _ => {
+                break;
+            }
+        }
+    }
+
+    let (num, unit) = property.split_at(indx);
+    match num.parse::<f64>() {
+        Ok(f) => Ok((f, unit)),
+        Err(_) => Err(()),
+    }
+}
+
+const UNIT: i64 = 1;
+const FILE_SIZE_MAGNITUDE: i64 = 1024;
+const FILE_SIZE_B: i64 = UNIT;
+const FILE_SIZE_KB: i64 = FILE_SIZE_B * FILE_SIZE_MAGNITUDE;
+const FILE_SIZE_MB: i64 = FILE_SIZE_KB * FILE_SIZE_MAGNITUDE;
+const FILE_SIZE_GB: i64 = FILE_SIZE_MB * FILE_SIZE_MAGNITUDE;
+// When we cast TB and PB to `f64`, compiler will report a overflow warning.
+// FILE_SIZE_TB = FILE_SIZE_GB * FILE_SIZE_MAGNITUDE
+// FILE_SIZE_PB = FILE_SIZE_PB * FILE_SIZE_MAGNITUDE
+
+const TIME_MAGNITUDE: i64 = 1000;
+const TIME_MS: i64 = UNIT;
+const TIME_S: i64 = TIME_MS * TIME_MAGNITUDE;
+
+#[allow(match_same_arms)]
+pub fn get_integer_by_string(size: &str) -> Result<i64, ()> {
+    let (num, unit) = try!(split_property(size));
+
+    match unit {
+        // file size
+        "B" => Ok((num * (FILE_SIZE_B as f64)) as i64),
```

Comment:
I think we should keep `B`. People may get confused as they can write something like "10KB", "1.5MB", but why not "50B"?